metadata.go

```diff
@@ -23,7 +23,7 @@ func (m Metadata) Equal(other Metadata) bool {
 }
 
 // EncodeMetadata serializes Metadata to []byte.
-func EncodeMetadata(m Metadata) []byte {
+func (m Metadata) Encode() []byte {
```

Comment:
Any particular reason to prefer object style over functional style here? I guess it doesn't matter and object style always reads easier to me.